### PR TITLE
v.hull: Add the test cases

### PR DESCRIPTION
**Pull request by codersmiti**
This PR introduces test suite for the v.hull GRASS GIS module, covering a range of scenarios, including validating the module's functionality, geometric correctness, and edge-case handling for convex hull generation. The test cases cover a variety of scenarios, including collinear points, non-collinear points, duplicate points, concave shapes, and 3D inputs.

**Key updates in this PR include:**

- **Functionality**: Validate convex hull creation for 2D and 3D point sets, ensuring correct output existence and dimensions.

- **Edge Cases**: Test collinear points, duplicate points, and concave shapes to ensure robust handling of degenerate and non-standard inputs.

- **Geometric Correctness**: Verify the number of vertices, polygon closure, and correct exclusion of interior concave points.

- **3D Inputs**: Validate the formation of a tetrahedron from 3D input points.

**Test Case Additions:**

- **Basic Functionality:**

     - Verify the creation and existence of the convex hull output.

     - Test input handling for 2D and 3D point sets.

- **Edge Cases:**

     - Collinear Points: Ensure the convex hull forms a degenerate boundary with exactly 3 vertices.

     - Duplicate Points: Test that duplicate points do not affect the hull structure.

    - Concave Shapes: Validate that the convex hull ignores interior concave points.

- **Geometric Correctness:**

     - Non-Collinear Points: Verify the convex hull correctly encloses a non-collinear set of points.

    - 3D Inputs: Test the formation of a tetrahedron from 3D input points.

### CODE REVIEW

File: vector/v.hull/testsuite/v_hull_test.py [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+        self.assertIsNotNone(geometry_line, "No valid geometry found in output")
+        parts = geometry_line.split()
+
+        self.assertEqual(parts[0], "B", "Output should be a boundary")
+        self.assertEqual(int(parts[1]), 3, "Boundary should have 3 vertices")
+        self.assertEqual(
+            parts[2:4],
+            parts[6:8],
+            "Third vertex should be a copy of the first vertex in collinear case",
+        )
```
Comment (petrasovaa) [inline vector/v.hull/testsuite/v_hull_test.py:120]: I think you can replace this (and the other tests) with https://grass.osgeo.org/grass-devel/manuals/libpython/gunittest.html#gunittest.case.TestCase.assertVectorFitsTopoInfo

File: raster/r.geomorphon/r.geomorphon.html
Comment (petrasovaa) [inline raster/r.geomorphon/r.geomorphon.html:1]: Please remove this from the PR.

File: vector/v.hull/testsuite/v_hull_test.py
Comment (petrasovaa) [inline vector/v.hull/testsuite/v_hull_test.py:1]: We use `test_v_hull.py` for test file names.

**Review by petrasovaa — CHANGES_REQUESTED** [review]

File: vector/v.hull/testsuite/v_hull_test.py [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+        """Cleans up test data by removing all created vector maps after tests complete."""
+        cls.runModule(
+            "g.remove",
+            type="vector",
+            name=f"{cls.input_points},{cls.output_hull},"
+            f"{cls.input_colinear},{cls.output_colinear_hull},"
+            f"{cls.input_non_colinear},{cls.output_non_colinear_hull},"
+            f"{cls.input_duplicate},{cls.output_duplicate},"
+            f"{cls.input_concave},{cls.output_concave_hull},"
+            f"{cls.input_3d},{cls.output_3d_hull}",
```
Comment (wenzeslaus) [inline vector/v.hull/testsuite/v_hull_test.py:35]: This can be a Python list

File: vector/v.hull/testsuite/v_hull_test.py
Comment (codersmiti) [reply vector/v.hull/testsuite/v_hull_test.py:1]: Renamed the file name to test_v_hull.py.

File: vector/v.hull/testsuite/v_hull_test.py (hunk 2 above)
Comment (codersmiti) [reply vector/v.hull/testsuite/v_hull_test.py]: Updated the code to use a Python list.

**Review by petrasovaa — APPROVED** [review]
